brave/src/main/java/brave/Tracer.java

```diff
@@ -44,7 +44,35 @@
  * @see Span
  * @see Propagation
  */
-public final class Tracer {
+public final class Tracer implements Closeable {
+  // volatile for visibility on get. writes guarded by Tracer.class
+  static volatile Tracer current = null;
+
+  /**
+   * Returns the most recent tracer that hasn't been closed.
```

Comment:
Maybe its better as:

`Returns the most recently created tracer iff it hasn't been closed. null otherwise